### ISSUE TITLE
Test Code

### ISSUE THREAD

**Issue by Vhauryn** [Vhauryn/Geoffray#7]
Write unit tests!

- [ ] implement a test environment
- [ ] write unit tests!

**Comment by stefanciprian** [comment]
Added dart:test and using dio http client to perform requests

**Comment by stefanciprian** [comment]
@ehildt We need to find a way to start/close a server when test runs. At the moment, I create a server for each test 👎 .

**Comment by ehildt** [comment]
> @ehildt We need to find a way to start/close a server when test runs. At the moment, I create a server for each test 👎 .

Yes. This is very true. We don't want to start a new server for every test! 
We also need to think of a way to run penetration tests against the server.

For now let's focus on testing those hooks that don't need a server to run.
We basically can test all the hooks that interact with the context or state etc.

Being honest i need more knowledge about the Dart VM in general.
